### PR TITLE
How to install tesseract on windows

### PR DESCRIPTION
**Pull request by LaurentBerger**
How to install tesseract on windows.

Please check my english!

### CODE REVIEW

File: modules/text/tutorials/install_tesseract/install_tesseract.markdown [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  git clone $2
+  mkdir Build/$1
+else
+  echo "update $1"
+  cd $1
+  git pull
+  cd ..
+fi
+}
+echo "Installation de leptonica et tesseract"
```
Comment (berak) [inline modules/text/tutorials/install_tesseract/install_tesseract.markdown]: echo "Installing leptonica and tesseract"

File: modules/text/tutorials/install_tesseract/install_tesseract.markdown [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+@endcode
+    and leptonica, tesseract will be installed in c:/lib/install
+@code{.bash}
+-DCMAKE_INSTALL_PREFIX="$myRepo"/install/"$RepoSource" "$myRepo/$RepoSource"
+@endcode
+    with no Perf tests, no tests, no doc, no CUDA and no example
+@code{.bash}
+CMAKE_OPTIONS='-DBUILD_PERF_TESTS:BOOL=OFF -DBUILD_TESTS:BOOL=OFF -DBUILD_DOCS:BOOL=OFF  -DWITH_CUDA:BOOL=OFF -DBUILD_EXAMPLES:BOOL=OFF'
+@endcode
+-#  In git command line enter following command :
```
Comment (berak) [inline modules/text/tutorials/install_tesseract/install_tesseract.markdown]: the following command :

File: modules/text/tutorials/table_of_content_text.markdown [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Text module {#tutorial_table_of_content_text}
+=====================
+
+-   @subpage tutorial_install_tesseract
+
+    *Compatibility:* \> OpenCV 3.4
+
+    *Author:* Laurent Berger
+
+    Instructions in order to properly setup tesseract on windows for text module.
```
Comment (berak) [inline modules/text/tutorials/table_of_content_text.markdown]: the text module

File: modules/text/tutorials/install_tesseract/install_tesseract.markdown [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+@code{.bash}
+./installTesseract.sh
+@endcode
+-# you will have to choose between clone https://github.com/tesseract-ocr/tessdata or only copy language file you need in a folder (example c:\\lib\\install\\tesseract\\tessdata). If you don't want to add a new folder you must copy language file in same folder than your executable
+-# if you created a new folder then you must add to system  a new variable TESSDATA_PREFIX wiih value c:\\lib\\install\\tessdata
+-# add to variable system PATH c:\\Lib\\install\\leptonica\\bin and c:\\Lib\\install\\tesseract\\bin. If you don't want to modify variable PATH then copy tesseract400.dll and leptonica-1.74.4.dll in same directory than your exe file
```
Comment (berak) [inline modules/text/tutorials/install_tesseract/install_tesseract.markdown]: add c:\\Lib\\install\\leptonica\\bin and c:\\Lib\\install\\tesseract\\bin to your PATH environment variable.

if you don't want to modify the PATH, copy tesseract400.dll and leptonica-1.74.4.dll to same directory than your exe file.

File: modules/text/tutorials/install_tesseract/install_tesseract.markdown [hunk 5]
```diff
@@ -0,0 +1,79 @@
+Tesseract (master) installation by Using git-bash (version>=2.14.1) and cmake (version >=3.9.1){#tutorial_install_tesseract}
+===============================================================
+
+-#  We suppose you installesd opencv and opencv_contrib using [this tutorials](http://docs.opencv.org/master/d3/d52/tutorial_windows_install.html#tutorial_windows_gitbash_build]
```
Comment (berak) [inline modules/text/tutorials/install_tesseract/install_tesseract.markdown]: we assume you installed opencv ...

File: modules/text/tutorials/install_tesseract/install_tesseract.markdown [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    with no Perf tests, no tests, no doc, no CUDA and no example
+@code{.bash}
+CMAKE_OPTIONS='-DBUILD_PERF_TESTS:BOOL=OFF -DBUILD_TESTS:BOOL=OFF -DBUILD_DOCS:BOOL=OFF  -DWITH_CUDA:BOOL=OFF -DBUILD_EXAMPLES:BOOL=OFF'
+@endcode
+-#  In git command line enter following command :
+@code{.bash}
+./installTesseract.sh
+@endcode
+-# you will have to choose between clone https://github.com/tesseract-ocr/tessdata or only copy language file you need in a folder (example c:\\lib\\install\\tesseract\\tessdata). If you don't want to add a new folder you must copy language file in same folder than your executable
+-# if you created a new folder then you must add to system  a new variable TESSDATA_PREFIX wiih value c:\\lib\\install\\tessdata
```
Comment (berak) [inline modules/text/tutorials/install_tesseract/install_tesseract.markdown]: if you created a new folder, then you must add a new variable, TESSDATA_PREFIX with the value c:\\lib\\install\\tessdata to your system's environment

File: modules/text/tutorials/install_tesseract/install_tesseract.markdown [hunk 7]
```diff
@@ -0,0 +1,79 @@
+Tesseract (master) installation by Using git-bash (version>=2.14.1) and cmake (version >=3.9.1){#tutorial_install_tesseract}
```
Comment (berak) [inline modules/text/tutorials/install_tesseract/install_tesseract.markdown]: using (lowercase)

File: modules/text/tutorials/install_tesseract/install_tesseract.markdown [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+@endcode
+    with no Perf tests, no tests, no doc, no CUDA and no example
+@code{.bash}
+CMAKE_OPTIONS='-DBUILD_PERF_TESTS:BOOL=OFF -DBUILD_TESTS:BOOL=OFF -DBUILD_DOCS:BOOL=OFF  -DWITH_CUDA:BOOL=OFF -DBUILD_EXAMPLES:BOOL=OFF'
+@endcode
+-#  In git command line enter following command :
+@code{.bash}
+./installTesseract.sh
+@endcode
+-# you will have to choose between clone https://github.com/tesseract-ocr/tessdata or only copy language file you need in a folder (example c:\\lib\\install\\tesseract\\tessdata). If you don't want to add a new folder you must copy language file in same folder than your executable
```
Comment (berak) [inline modules/text/tutorials/install_tesseract/install_tesseract.markdown]: now we need the language files from tesseract. either clone https://github.com/tesseract-ocr/tessdata, or copy only those language files you need to a folder (example c:\\lib\\install\\tesseract\\tessdata).

File: modules/text/tutorials/install_tesseract/install_tesseract.markdown (hunk 1 above)
Comment (LaurentBerger) [reply modules/text/tutorials/install_tesseract/install_tesseract.markdown]: OK

File: modules/text/tutorials/install_tesseract/install_tesseract.markdown (hunk 2 above)
Comment (LaurentBerger) [reply modules/text/tutorials/install_tesseract/install_tesseract.markdown]: OK

File: modules/text/tutorials/install_tesseract/install_tesseract.markdown (hunk 4 above)
Comment (LaurentBerger) [reply modules/text/tutorials/install_tesseract/install_tesseract.markdown]: OK

File: modules/text/tutorials/install_tesseract/install_tesseract.markdown (hunk 5 above)
Comment (LaurentBerger) [reply modules/text/tutorials/install_tesseract/install_tesseract.markdown]: OK

File: modules/text/tutorials/install_tesseract/install_tesseract.markdown (hunk 7 above)
Comment (LaurentBerger) [reply modules/text/tutorials/install_tesseract/install_tesseract.markdown]: OK

File: modules/text/tutorials/install_tesseract/install_tesseract.markdown (hunk 8 above)
Comment (LaurentBerger) [reply modules/text/tutorials/install_tesseract/install_tesseract.markdown]: OK

File: modules/text/tutorials/table_of_content_text.markdown (hunk 3 above)
Comment (LaurentBerger) [reply modules/text/tutorials/table_of_content_text.markdown]: OK

**Comment by LaurentBerger** [comment]
Don't merge It's wrong

**Comment by alalek** [comment]
@LaurentBerger Thank you for the tutorial!
@berak Thank you for checks!

:+1:

**Pull request merged**